site/content/docs/03-template-syntax.md

```diff
@@ -1101,6 +1091,32 @@ A custom transition function can also return a `tick` function, which is called
 
 If a transition returns a function instead of a transition object, the function will be called in the next microtask. This allows multiple transitions to coordinate, making [crossfade effects](/tutorial/deferred-transitions) possible.
 
+Transition functions also receive a third argument, `options`, which contains information about the transition.
+
+Available values in the `options` object are:
+
+* `direction` - one of `in`, `out`, or `both` depending on the type of transition
+
+```sv
+<script>
+	export let visible = false;
+
+	function attribute(node, params, options) {
+		node.dataset.direction = options.direction;
```

Comment:
I don't think this is a real-world example.
(I didn't get `node.dataset.direction` will use for what?)

For example, I think we will add the below example.
(But I believe you are more familiar with the specific use cases than I am.)

```js
export function my_transition(node, params, options) {
	const { direction } = options;
	if (direction === 'in') return wacky_intro_params(params);
	if (direction === 'out') return wacky_outro_params(params);
	return coordinated_outro(params);
}
```

And according to other examples, these should work after just copying and pasting.

Reply:
My real-world usecase is a nice embedded browser that doesn't support generating `@keyframes` and injecting them dynamically, so I need to either use a pre-set `@keyframes` definition using css variables for the `from`/`to` values *or* I can inject inline styles to set up a transition, wait for a tick, and then change the style to trigger the transition.

Neither of which is super easy to do in a small example, so I was trying to do something extremely short to get the idea across.

I can make it do *something* when you copy and paste but it's gonna bloat out the example a bit 🫤

Reply:
Humm. I also thought example but I need 18 lines.
Then I think we don't need to show examples in the docs at least for now.
(Sorry for changing what I said🙏)

```svelte
<script>
	import { slide, fade } from "svelte/transition";
	let visible;
	function custom_transition(node, params, options) {
		const { direction } = options;
		if (direction === "in") return () => slide(node, params);
		if (direction === "out") return () => fade(node, params);
		return null
	}
</script>

<button on:click={() => (visible = !visible)}>Click Me</button>
{#if visible}
	<div in:custom_transition={{ duration: 1000 }}>in</div>
{/if}
{#if visible}
	<div out:custom_transition={{ duration: 500 }}>out</div>
{/if}

```